Source/spells.cpp

```diff
@@ -296,46 +293,37 @@ void DoResurrect(int pnum, int rid)
 
 void DoHealOther(int pnum, int rid)
 {
-	int i, j, hp;
-
 	if (pnum == myplr) {
 		NewCursor(CURSOR_HAND);
 	}
 
 	auto &player = plr[pnum];
 	auto &target = plr[rid];

```

Comment:
`rid` should be checked before using it to index into the `plr` array. This causes OOB reads if an invalid network packet is received.

Reply:
Doing so would also get rid of the `uint16 -1` madness.